cmake/MiscCheck.cmake

```diff
@@ -159,3 +159,15 @@ endif()
 
 # ---[ Create CAFFE2_BUILD_SHARED_LIBS for macros.h.in usage.
 set(CAFFE2_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
+
+# ---[ Check if we will need to include the local Modules_CUDA_fix folder.
+# Add your conditions here if needed.
+if (MSVC)
```

Comment:
Recently I found that with this diff, we need `set(CAFFE2_CMAKE_USE_LOCAL_FINDCUDA ON)` for linux systems with cmake <3.7 to avoid the nvcc error. Maybe add a cmake version check here too?